src/librustc_codegen_llvm/builder.rs

```diff
@@ -1248,7 +1248,8 @@ impl Builder<'a, 'll, 'tcx> {
 
         let lifetime_intrinsic = self.cx.get_intrinsic(intrinsic);
 
-        let ptr = self.pointercast(ptr, self.cx.type_i8p());
+        let ptr = self
+            .pointercast(ptr, self.cx.type_i8p(self.cx.data_layout().instruction_address_space));
```

Comment:
TODO: this can be turned into a "preserving" address space from `ptr` rather than throwing it away because it _should_ be the same as `self.cx.data_layout().instruction_address_space)`